.github/workflows/appsignals-e2e-eks-test.yml

```diff
@@ -219,6 +220,25 @@ jobs:
             fi
           done
 
+      - name: Install AppSignals addon
+        id: install-addon
+        if: steps.deploy-sample-app.outcome == 'success' && !cancelled()
+        run: |
+          source ${{ env.TEST_RESOURCES_FOLDER }}/.github/workflows/util/execute_and_retry.sh
+          execute_and_retry 2 \
+          "${{ env.TEST_RESOURCES_FOLDER }}/enablement-script/scripts/eks/appsignals/enable-app-signals.sh \
+          ${{ inputs.test-cluster-name }} \
+          ${{ inputs.aws-region }} \
+          ${{ env.SAMPLE_APP_NAMESPACE }}" \
+          "${{ env.TEST_RESOURCES_FOLDER }}/enablement-script/scripts/eks/appsignals/clean-app-signals.sh \
+          ${{ inputs.test-cluster-name }} \
+          ${{ inputs.aws-region }} \
+          ${{ env.SAMPLE_APP_NAMESPACE }} && \
+          aws eks update-kubeconfig --name ${{ inputs.test-cluster-name }} --region ${{ inputs.aws-region }}"
+          
+          kubectl delete pods --all -n ${{ env.SAMPLE_APP_NAMESPACE }}
+          kubectl wait --for=condition=Ready pod --all -n ${{ env.SAMPLE_APP_NAMESPACE }}
+
```

Comment:
We're restarting the pods, so we should repeat the endpoint check logic here. The code before this in the deployment step guarantees the deployment succeeded and the endpoint is reachable but then we restart the pods. Not sure of the likelihood that restarting pods will cause the service to not be reachable again.

If we still see transient failures due to endpoint not providing an appropriate response after this PR, this is where to start looking.

Reply:
If that is the case, moved the enable appsignal part back to the deploy terraform step.

Reply:
With additional retry logic